python/lsst/ts/hvac/mqtt_info_reader.py

```diff
@@ -0,0 +1,467 @@
+)
+dat_control_json_filename = input_dir.joinpath(
+    "JSON_V7_PUBLICACIONES_SUSCRIPCIONES.json"
+)
+
+
+class MqttInfoReader:
+    def __init__(self):
+        # This dict contains the general MQTT topics (one for each sub-system)
+        # as keys and a dictionary representing the items as value. The
+        # structure is
+        # "general_topic" : {
+        #     "item": {
+        #         "idl_type": idl_type,
+        #         "unit": unit,
+        #         "topic_type": topic_type,
+        #         "limits": (lower_limit, upper_limit),
+        #     }
+        # }
+        self.hvac_topics = {}
+        # This dict contains the general ts_xml telemetry topics (retrieved by
+        # using the `HvacTopic` enum) as keys and a dictionary representing the
+        # ts_xml telemetry items (retrieved by using the `TelemetryItem` enum)
+        # as values. The structure is
+        # "HvacTopic" : {
+        #     "TelemetryItem": {
+        #         "idl_type": idl_type,
+        #         "unit": unit,
+        #     }
+        # }
+        self.telemetry_topics = {}
+        # This dict contains the general ts_xml command topics (retrieved by
+        # using the `HvacTopic` enum) as keys and a dictionary representing the
+        # ts_xml command items (retrieved by using the `CommandItem` enum) as
+        # values. The structure is
+        # "HvacTopic" : {
+        #     "CommandItem": {
+        #         "idl_type": idl_type,
+        #         "unit": unit,
+        #     }
+        # }
+        self.command_topics = {}
+
+    def _determine_unit(self, unit_string):
+        """Convert the provided unit string to a string representing the unit.
+
+        Parameters
+        ----------
+        unit_string: `str`
+            The unit as read from the input file.
+
+        Returns
+        -------
+        unit: `str`
+            A string representing the unit.
+        """
+        return {
+            "-": "unitless",
+            "": "unitless",
+            "°C": "deg_C",
+            "bar": "bar",
+            "%": "%",
+            "hr": "h",
+            "%RH": "%",
+        }[unit_string]
+
+    def _parse_limits(self, limits_string):
+        """Parse the string value of the limits column by comparing it to known
+        regular expressions and extracting the minimum and maximum values.
+
+        Parameters
+        ----------
+        limits_string: `str`
+            The string containing the limits to parse.
+
+        Returns
+        -------
+        lower_limit: `float`
+            The lower limit
+        upper_limit: `float`
+            The upper limit
+
+        Raises
+        ------
+        ValueError
+            In case an unknown string pattern is found in the limits column.
+
+        """
+        lower_limit = None
+        upper_limit = None
+
+        match = re.match(
+            r"^(-?\d+)(/| a |% a |°C a | bar a |%RH a )(-?\d+)(%|°C| bar| hr|%RH)?$",
+            limits_string,
```

Comment:
It would be a bit more efficient to compile the regexes once in the constructor and use the compiled versions here. But if this runs quickly enough then there is no need, and having them here is a bit more readable.

Reply:
I'll leave it as is.